tidy3d/components/medium.py

```diff
@@ -5112,7 +5118,7 @@ def _weighted_avg(
                 return med
             else:
                 raise ValidationError("Invalid medium type for the components of 'Medium2D'.")
-            poles += [(a, weight * c) for (a, c) in pole_res.poles]
+            poles += [(a, weight * c) for (a, c) in pole_res.poles if c != 0.0]
```

Comment:
I also added this filtering behavior for the poles. Otherwise, a lot of poles are appended with c=0 get added which I think does nothing